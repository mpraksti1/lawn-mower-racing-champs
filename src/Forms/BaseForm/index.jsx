import React from 'react';
import StyledForm from './styles';

export default props => {
  return (
    <StyledForm>
      {props.children}
    </StyledForm>
  )
}
