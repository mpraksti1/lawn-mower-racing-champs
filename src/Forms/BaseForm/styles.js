import styled from 'styled-components';

const StyledForm = styled.div`
  input {
    font-size: 12px;
    height: 35px;
    padding: 16px;
    width: 100%;
    border: none;
    border-radius: 5px;
    box-shadow: 2px 2px 2px rgba(0,0,0,0.3);

    &::placeholder {
      color: ${({ theme }) => theme.colors.gray};
    }
  }

  select {
    width: 100%;
    background: #fefffe;
    border-radius: 5px;
    border: none;
    height: 36px;
    text-align: center;
    padding: 0px 16px;
    box-shadow: 2px 2px 2px rgba(0,0,0,0.3);
  }

  form {
    display: grid;
    grid-row-gap: 16px;
  }

  .icon-input {
    position: relative;

    svg {
      position: absolute;
      top: 50%;
      right: 10px;
      transform: translateY(-50%);
    }
  }
`;

StyledForm.displayName = 'BaseFormStyled';

export default StyledForm;
