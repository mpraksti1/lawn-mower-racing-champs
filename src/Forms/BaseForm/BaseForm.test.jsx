import BaseForm from '.';

describe('Base form component', () => {
  it('renders without crashing', () => {
    shallow(
      <BaseForm>
        <form>
          <p>A form goes here</p>
        </form>
      </BaseForm>,
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BaseForm>
        <form>
          <p>A form goes here</p>
        </form>
      </BaseForm>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
