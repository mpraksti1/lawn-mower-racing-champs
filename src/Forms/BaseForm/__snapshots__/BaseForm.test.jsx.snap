// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base form component matches snapshot 1`] = `
<BaseFormStyled>
  <form>
    <p>
      A form goes here
    </p>
  </form>
</BaseFormStyled>
`;
